### ISSUE TITLE
Dynamically set variables used in setups/teardowns cause failures in dry-run

### ISSUE THREAD

**Issue by Thonuck** [robotframework/robotframework#2489]
If I start the script with a normal pybot call, it is passing.
But if i'm calling it with argument --dryrun, it fails
Used pybot version is Robot Framework 3.0 (Python 2.7.3 on linux2)

```

*** Settings ***
Test Template    My Test Case Template

*** Testcases ***
Check it    1

*** Keywords ***
My Test Case Template
    [Arguments]
    ...     ${tcId}
    Log To Console    just some content to run
    [Teardown]   Very Special Teardown ${tcId}

Very Special Teardown 1
    Log To Console    \n=> Test Case Teardown
```

**Comment by pekkaklarck** [comment]
How does it fail? Also, could you try formatting the example in Markdown as a codeblock to make it easier to understand?

**Comment by Thonuck** [comment]
If the test is run with pybot and the --dryrun argument, the test fails with the output
`Variable '${tcId}' not found.`
But if the test is run without --dryrun, it is passing. With older version (tested only with Robot 2.8.7) both, dryrun and normal run is passing. See below the output of the two different calls:
![teardown_failing_real_run_passing](https://cloud.githubusercontent.com/assets/24457792/21027194/08717ad0-bd90-11e6-83bb-c17caa550d9b.png)

I'd expect, that the dryrun is also passing, if the "normal" test run is passing

**Comment by pekkaklarck** [comment]
The problem is that your teardown contains a variable that is set dynamically based on keyword argument and in dry-run variables are not set dynamically at all. The simplest example demonstrating the same problem is this:

```robotframework
*** Test Cases ***
Example
    ${kw} =    Set variable    No operation
    [Teardown]   ${kw}
```

Handling variables better in dry-run would be great but it would also be a huge task. A lot simple solution is just ignoring errors if teardown (or setup) contains variable that doesn't exist.

**Comment by pekkaklarck** [comment]
Dynamically setting variable to control which setup/teardown is rare enough that I consider this low priority fix in the end. Using variables defined in the settings table actually works and that's a more common use case for variables in setups/teardowns. Setting a default value for a variable in the settings table is also a workaround for this issue.

**Comment by pekkaklarck** [comment]
This was fixed in RF 3.0.1 so you shouldn't need any workarounds with the latest versions.
